src/PRAS2PowerSystems.jl

```diff
@@ -1,240 +1,265 @@
 
-        file_name = dir_name * "/$i.json"
-        PSY.IS.to_json(psy_sys, file_name, force=false)
-        @info "Succesfully exported PSY System with outage profile for scenario $(i)."
-    end
-    @info "Succesfully exported PSY System(s). The .json files and corresponding data .h5 files for all scenarios are available here : $(dir_name)."
+# Arguments
+
+  - `sys::PSY.System`: PowerSystems.jl system model
+  - `template::RATemplate`: PRAS problem template
+  - `method::PRASCore.SequentialMonteCarlo`: Simulation method to use
+
+# Returns
+
+ - PSY System with PSY.TimeSeriesForcedOutage for all components for which asset status is available
+"""
+function generate_outage_profile!(
+    sys::PSY.System,
+    template::RATemplate,
+    method::PRASCore.SequentialMonteCarlo,
+)
+    pras_system = generate_pras_system(sys, template)
+    resultsspecs = get_outage_pras_resultspec(template)
+    results = PRASCore.assess(pras_system, method, resultsspecs...)
+    add_asset_status!(sys, results, template)
+    return sys
 end
 
 """
-    generate_outage_profile(pras_system,num_runs,psy_sys,num_scenarios,location)
+    $(TYPEDSIGNATURES)
 
-Process the assess results to get timeseries of generator status and include
-this timeseries data to the corresponding component in PSY System and exported
-using to_json method (serializing the PSY System).
+Analyze resource adequacy using Monte Carlo simulation and add the asset status from the worst sample
+to PSY.TimeSeriesForcedOutage of the component.
 
-...
+Uses default template with PSY.Area AggregationTopology.
 
 # Arguments
 
-  - `pras_system::PRASCore.SystemModel`: PRAS System
-  - `num_runs::Int64`: Number of PRAS runs
-  - `psy_sys::PSY.System`: PSY System
-  - `num_scenarios::Int64`: Number of scenarios of user interest.
-  - `location::String`: Location to store outage profile.
-    ...
-
-# Examples
-
-```julia-repl
-julia> generate_outage_profile(results, pras_sys, psy_sys, 1)
-PSY System exported using to_json method in InfrastructureSystems
-```
-"""
-function generate_csv_outage_profile(
-    pras_system::PRASCore.SystemModel;
-    location::Union{Nothing, String}=nothing,
-    num_runs::Union{Nothing, Int}=nothing,
-    num_scenarios::Union{Nothing, Int}=nothing,
-)
+  - `sys::PSY.System`: PowerSystems.jl system model
+  - `method::PRASCore.SequentialMonteCarlo`: Simulation method to use
 
-    #kwargs handling
-    if (location === nothing)
-        location = dirname(dirname(@__DIR__))
-        @warn "Location to save the exported PSY System not specified. Using the data folder of the module."
-    end
+# Returns
 
-    if (num_runs === nothing)
-        num_runs = 10000
-        @warn "Number of samples of PRAS Assess not specified. Using 10,000 samples."
-    end
+    - PSY System with PSY.TimeSeriesForcedOutage for all components for which asset status is available
+"""
+function generate_outage_profile!(sys::PSY.System, method::PRASCore.SequentialMonteCarlo)
```

Comment:
This one should also call the `generate_outage_profile!` with RATemplate.

Reply:
DOne.